pkg/source/source.go

```diff
@@ -58,54 +54,64 @@ type SyncingSource interface {
 }
 
 // Kind creates a KindSource with the given cache provider.
-func Kind(cache cache.Cache, object client.Object) SyncingSource {
-	return &internal.Kind{Type: object, Cache: cache}
+func Kind[T client.Object](cache cache.Cache, object T, handler handler.TypedEventHandler[T], predicates ...predicate.TypedPredicate[T]) SyncingSource {
+	return &internal.Kind[T]{
+		Type:       object,
+		Cache:      cache,
+		Handler:    handler,
+		Predicates: predicates,
+	}
 }
 
-var _ Source = &Channel{}
+var _ Source = &channel[string]{}
 
 // Channel is used to provide a source of events originating outside the cluster
 // (e.g. GitHub Webhook callback).  Channel requires the user to wire the external
 // source (e.g. http handler) to write GenericEvents to the underlying channel.
-type Channel struct {
+func Channel[T any](source <-chan event.TypedGenericEvent[T], handler handler.TypedEventHandler[T], predicates ...predicate.TypedPredicate[T]) Source {
+	return &channel[T]{
+		source:     source,
+		handler:    handler,
+		predicates: predicates,
+	}
+}
+
+type channel[T any] struct {
 	// once ensures the event distribution goroutine will be performed only once
 	once sync.Once
 
-	// Source is the source channel to fetch GenericEvents
-	Source <-chan event.GenericEvent
+	// source is the source channel to fetch GenericEvents
+	source <-chan event.TypedGenericEvent[T]
 
-	// dest is the destination channels of the added event handlers
-	dest []chan event.GenericEvent
+	handler handler.TypedEventHandler[T]
+
+	predicates []predicate.TypedPredicate[T]
 
```

Comment:
Would it make sense to preserve this option?

Reply:
The reason I ended up not preserving it is that I unexported source.Channel and exported a constructor for it instead. That was done to avoid silently (as in: Without compilation failures) breaking ppl who do not pass in a `Handler` which is now mandatory (it was also mandatory previously, but passed in through `Start()`).

The constructor func already uses its variadic arg for `Predicates` as they are optional. This setting would also be optional, so in order to make that happen, we would need to make the variadic arg some form of `ChannelOption` and that would make it more annoying to pass in `Predicates`. I assumed that this is rarely if ever configured and thus that its best to use the variadic arg for the `Predicates`. I don't have any evidence one way or the other though and not strong take, happy to change this if you think we should keep this configurable.